frontend/src/components/MessagesPage.vue

```diff
@@ -1,36 +1,33 @@
 <template>
   <v-container id="messages" fluid class="pa-0 ma-0" height="100%">
-    <v-row v-if="!allMessages" class="ma-0">
+    <v-row v-if="!allMessages || allMessages.length === 0" class="ma-0">
```

Comment:
with this change, if the provider doesn't have any messages, the page will stuck at loading (spinner) screen. The spinner should only be displayed when we're loading the messages. If the provider doesn't have any message, we can display No data message (v-data-table should show that text by default if it is empty)

Reply:
we can add a loading variable. set it to true before we call getAllMessages, then set it to false after it's complete. (in mounted)